### PR TITLE
LPC11xx experimental branch

### PR DESCRIPTION
**Pull request by farcaller**
This is work in progress of my attempt to an architecture based purely on the placement ioregs (the reason why it's not lpc176x getting a refab is that this mcu has just 20 register blocks and is rather trivial).

The ideas I want to pinpoint are:

 * `mod regs` is private, any access is done via relevant APIs that are `unsafe` when reasonable
 * new way to invoke ioreg tests (swapped a library for one with simpler extensions and better output)

### CODE REVIEW

File: src/hal/lpc11xx/syscon.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    let j = thread::Builder::new()
+        .name(String::from("fails_to_set_isr_location_to_bootloader"))
+        .spawn(|| {
+      init_replayer!();
+      expect_volatile_write!(0x4004_8000, 0);
+      set_isr_location(ISRLocation::Bootloader);
+    }).unwrap();
+    let res = j.join();
+
+    expect!(res.is_err()).to(be_equal_to(true));
```
Comment (zummenix) [inline src/hal/lpc11xx/syscon.rs]: I think it can be:
```rust
expect!(res).to(be_err());
```

Comment (farcaller) [inline src/hal/lpc11xx/syscon.rs]: Yup, apart from the issue that it was a compiler error for some reason :smile:

Comment (mcoffin) [inline src/hal/lpc11xx/syscon.rs]: > compiler error for some reason

Two hip-hip hoorays for having to track nightlies!

Comment (farcaller) [inline src/hal/lpc11xx/syscon.rs]: > Two hip-hip hoorays for having to track nightlies!

Did I tell you that we will need to fork cargo to keep the nightly going? Buy the ticket, take the ride! :roller_coaster:

Comment (niclashoyer) [inline src/hal/lpc11xx/syscon.rs]: > we will need to fork cargo
because of rust-lang/cargo#1796 ?

**Comment by farcaller** [comment]
This fixes lots of things apart form lpc11xx so I will land it now and deal with consequences later. LPC11xx is still non-existent and isn't expected to work meaningfully.

**Pull request merged**